### PR TITLE
cmake: flash: Update WEST_DIR path

### PR DESCRIPTION
**Pull request by eHammarstrom**
See https://docs.zephyrproject.org/latest/guides/west/without-west.html#flashing-and-debugging for more context.

Change based off current state of west/master.

Signed-off-by: Emil Hammarstrom <emil.a.hammarstrom@gmail.com>

### CODE REVIEW

**Review by carlescufi — COMMENTED** [review]
LGTM, but I wonder if we should check the west version and conditionally use `src/west/` or `src/west/app/`

File: cmake/flash/CMakeLists.txt [hunk 1]
```diff
@@ -167,7 +167,7 @@ if(DEFINED ENV{WEST_DIR} AND NOT WEST_DIR)
 endif(DEFINED ENV{WEST_DIR} AND NOT WEST_DIR)
 
 if(WEST_DIR)
-  set(WEST "PYTHONPATH=${WEST_DIR}/src" "${PYTHON_EXECUTABLE};${WEST_DIR}/src/west/main.py;--zephyr-base=$ENV{ZEPHYR_BASE} ")
+  set(WEST "PYTHONPATH=${WEST_DIR}/src" "${PYTHON_EXECUTABLE};${WEST_DIR}/src/west/app/main.py;--zephyr-base=$ENV{ZEPHYR_BASE} ")
```
Comment (mbolivar) [inline cmake/flash/CMakeLists.txt]: Can you update the `west>=0.6.2` line in scripts/requirements.txt also? To say e.g. `>=0.7.2`, the current release.

**Review by mbolivar — COMMENTED** [review]
Thanks for this. Just one request.

File: cmake/flash/CMakeLists.txt (hunk 1 above)
Comment (eHammarstrom) [reply cmake/flash/CMakeLists.txt]: Sure.

Comment (SebastianBoe) [inline cmake/flash/CMakeLists.txt]: Does this change require us to bump the MIN_WEST_VERSION in host-tools.cmake ?

Comment (mbolivar) [reply cmake/flash/CMakeLists.txt]: Good catch, yes.

Comment (eHammarstrom) [reply cmake/flash/CMakeLists.txt]: A quick glance seem to indicate that versions >= 0.7.0 has this structure. Can the MIN_WEST_VERSION be bumped to 0.7.0?

Comment (mbolivar) [reply cmake/flash/CMakeLists.txt]: I think 0.7.2 has enough bugfixes compared to 0.7.0 that it's worth switching to it as a minimum if we are going to drop support for 0.6.x.

**Comment by eHammarstrom** [comment]
@mbolivar The "Shippable" CI fails because it uses West 0.7.1, how do we resolve this?

**Comment by mbolivar** [comment]
> @mbolivar The "Shippable" CI fails because it uses West 0.7.1, how do we resolve this?

Sigh. Let's go with 0.7.1 for now then. I'll try to get the container updated at some point later -- @galak is on vacation right now and he manages the container releases these days.

**Comment by eHammarstrom** [comment]
> > @mbolivar The "Shippable" CI fails because it uses West 0.7.1, how do we resolve this?
> 
> Sigh. Let's go with 0.7.1 for now then. I'll try to get the container updated at some point later -- @galak is on vacation right now and he manages the container releases these days.

Ok! Fixed.

**Comment by eHammarstrom** [comment]
ping @mbolivar

File: cmake/host-tools.cmake [hunk 2]
```diff
@@ -38,7 +38,7 @@ else()
   # even after output is one line.
   message(STATUS "Found west: ${WEST} (found suitable version \"${west_version}\", minimum required is \"${MIN_WEST_VERSION}\")")
 
-  if (${west_version} VERSION_GREATER_EQUAL "0.7.1")
+  if (${west_version} VERSION_GREATER_EQUAL ${MIN_WEST_VERSION})
```
Comment (SebastianBoe) [inline cmake/host-tools.cmake]: This is always true isn't it? Due to us error-ing out above when it is false ?

Comment (eHammarstrom) [reply cmake/host-tools.cmake]: Yes, `FATAL_ERROR` does exit after printing the message. I'll remove the check.

**Comment by eHammarstrom** [comment]
ping @SebastianBoe

**Review by SebastianBoe — APPROVED** [review]

**Comment by carlescufi** [comment]
@mbolivar-nordic ok with merging this?
@eHammarstrom can you please rebase?

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by eHammarstrom** [comment]
ping @mbolivar-nordic @carlescufi

**Pull request merged**